loadbalancer/algorithm.go

```diff
@@ -55,20 +55,36 @@ func fadeIn(now time.Time, duration time.Duration, exponent float64, detected ti
 	return math.Pow(float64(rel)/float64(duration), exponent)
 }
 
-func shiftToOldest(ctx *routing.LBContext) routing.LBEndpoint {
+func shiftWeighted(rnd *rand.Rand, ctx *routing.LBContext, w []float64, now time.Time) routing.LBEndpoint {
+	var sum float64
+	weightSums := w
+	rt := ctx.Route
 	ep := ctx.Route.LBEndpoints
-	oldest := ep[0]
-	for _, epi := range ep[1:] {
-		if epi.Detected.Before(oldest.Detected) {
-			oldest = epi
+	for _, epi := range ep {
+		wi := fadeIn(now, rt.LBFadeInDuration, rt.LBFadeInExponent, epi.Detected)
+		sum += wi
+		weightSums = append(weightSums, sum)
+	}
+
+	var choice routing.LBEndpoint
```

Comment:
Wouldn't `choice := ep[len(ep)-1]` eliminate `if i == len(weightSums)-1` in the loop?

Reply:
true, thx!